### PR TITLE
[Cloud Sync] Merge remote-tracking branch 'origin/master' 2021-07-19

### PR DESCRIPTION
**Pull request by isacikgoz**
#### Summary
Merge master into cloud branch for cloud release

#### Release Note

```release-note
NONE
```

### CODE REVIEW

File: store/sqlstore/upgrade.go [hunk 1]
```diff
@@ -1334,27 +1336,114 @@ func hasMissingMigrationsVersion536(sqlStore *SqlStore) bool {
...
-	sqlStore.RemoveIndexIfExists("idx_teams_name", "Teams")
-	sqlStore.RemoveIndexIfExists("idx_user_access_tokens_token", "UserAccessTokens")
-	sqlStore.RemoveIndexIfExists("idx_user_terms_of_service_user_id", "UserTermsOfService")
-	sqlStore.RemoveIndexIfExists("idx_users_email", "Users")
-	sqlStore.RemoveIndexIfExists("idx_sharedchannelusers_user_id", "SharedChannelUsers")
-	sqlStore.RemoveIndexIfExists("IDX_RetentionPolicies_DisplayName_Id", "RetentionPolicies")
-	sqlStore.CreateIndexIfNotExists("IDX_RetentionPolicies_DisplayName", "RetentionPolicies", "DisplayName")
-
-	// saveSchemaVersion(sqlStore, Version5370)
+	if shouldPerformUpgrade(sqlStore, Version5360, Version5370) {
```
Comment (streamer45) [inline store/sqlstore/upgrade.go:1339]: Do we need to provide `hasMissingMigrationsVersion537` to handle partial migrations? That's usually been the pattern with cloud releases.

File: store/sqlstore/upgrade.go [hunk 2]
```diff
@@ -19,7 +19,8 @@ import (
 )
 
 const (
-	CurrentSchemaVersion   = Version5360
+	CurrentSchemaVersion   = Version5370
```
Comment (streamer45) [inline store/sqlstore/upgrade.go:22]: Did we recently change how we set versions? I thought we were setting this to the newest version but now I see we are in sync with the on-prem release. Of course the issue remains that whatever we set here is not 100% reflective of the schema state.

File: store/sqlstore/upgrade.go (hunk 1 above)
Comment (isacikgoz) [reply store/sqlstore/upgrade.go:1339]: Hmm doesn't the `CRTChannelMembershipCountsMigrationComplete` and `CRTThreadCountsAndUnreadsMigrationComplete` values do this? I think it wasn't needed to check if migration is required if something like these checks are provided.

File: store/sqlstore/upgrade.go (hunk 2 above)
Comment (isacikgoz) [reply store/sqlstore/upgrade.go:22]: This info probably escaped from me, I don't remember a recent decision change about versioning as well. Going to update the version to `5380`. Then we should also save the schema version in `upgradeDatabaseToVersion538` right?

Comment (streamer45) [reply store/sqlstore/upgrade.go:22]: Yes, from what I recall that's what we used to do. But first let's see if @jwilander has any further input on this.

File: store/sqlstore/upgrade.go (hunk 1 above)
Comment (streamer45) [reply store/sqlstore/upgrade.go:1339]: I was actually looking at 5.37 changes. I guess we should be okay considering the version wasn't bumped until now.

Comment (isacikgoz) [reply store/sqlstore/upgrade.go:1339]: Ah.. yeah 🤦

File: store/sqlstore/upgrade.go (hunk 2 above)
Comment (jwilander) [reply store/sqlstore/upgrade.go:22]: There was no decision to change how we did it so we should set it to the newest version and save the schema version in `upgradeDatabaseToVersion538`

**Review by jwilander — APPROVED** [review]

**Review by streamer45 — APPROVED** [review]

**Pull request merged**